File: maxSubArray.py
def maxSubArray(n):
	total_sum=max_sum= n[0]

	for i in n[1:]:
		total_sum=max(i,total_sum+i)
		max_sum=max(max_sum,total_sum)
	return max_sum

            
n = [-2,1,-3,4,-1,2,1,-5,4]
x=maxSubArray(n)
print (x)

File: maxSubArray.java
class maxSubArray{
	public static void main(String [] args){
		int [] nums = {-2,1,-3,4,-1,2,1,-5,4};
		int total_sum=nums[0];
		int max_sum=nums[0];
		for( int n : nums){
			total_sum=Math.max(n,total_sum+n);
			max_sum=Math.max(max_sum,total_sum);

	}
	System.out.println(max_sum);
	}

}